application/core/LSYii_ControllerRest.php

```
@@ -0,0 +1,244 @@
+        } catch (Exception $e) {
+            $this->displayException($e);
+        }
+    }
+
+    /**
+     * Return data to browser as JSON with the correct hHTTPttp response code.
+     *
+     * @param Response $response
+     * @return void
+     */
+    protected function renderCommandResponse(Response $response)
+    {
+        $this->renderJSON(
+            $response->getData(),
+            $this->getHttpResponseCode($response->getStatus())
+        );
+    }
+
+    /**
+     * Return data to browser as JSON and end application.
+     *
+     * @param array $data
+     * @param int $responseCode
+     * @return void
+     */
+    protected function renderJSON($data, $responseCode = 200)
+    {
+        http_response_code($responseCode);
+        header('Content-type: application/json');
+        echo CJSON::encode($data);
+
+        foreach (Yii::app()->log->routes as $route) {
+            if ($route instanceof CWebLogRoute) {
+                $route->enabled = false; // disable any weblogroutes
+            }
+        }
+        Yii::app()->end();
+    }
+
+    /**
+     * Get HTTP response code from command response status.
+     *
+     * @param StatusAbstract $status
+     * @return void
+     */
+    protected function getHttpResponseCode(StatusAbstract $status)
+    {
+        $httpCode = 200;
+        switch ($status->getCode()) {
+            case 'success':
+                $httpCode = 200;
+                break;
+            case 'success_created':
+                $httpCode = 201;
+                break;
+            case 'error':
+                $httpCode = 400;
+                break;
+            case 'error_unauthorised':
+                $httpCode = 401;
+                break;
+            case 'error_bad_request':
+                $httpCode = 400;
+                break;
+            case 'error_not_found':
+                $httpCode = 404;
+                break;
+        }
+        return $httpCode;
```

200 should not be the default, I think. Rather throw an exception?